libs/js-sdk/src/utils/retry.ts

```diff
@@ -20,7 +22,15 @@ export async function retry<T>(
             // Max retries
             throw error
         }
-        const delayMs = initialDelayMs * Math.pow(backoffFactor, retryAttempt)
+        if (error instanceof TooManyRequestsError && error.retryAfter > new Date()) {
+            retryAfterMs = error.retryAfter.getTime() - new Date().getTime()
+        }
+        const delayMs = initialDelayMs * Math.pow(backoffFactor, retryAttempt) + retryAfterMs
```

Comment:
I think we would want to use retryAfterMs as delayMs if we handle the error.

Otherwise we could back off a few times, and wait far longer than we have to to retry?